File: src/imu_pkg/src/imu_node.cpp
#include "ros/ros.h"
#include "sensor_msgs/Imu.h"
#include "tf/tf.h"
#include "geometry_msgs/Twist.h"

ros::Publisher vel_pub;

void imu_callback(const sensor_msgs::Imu msg)
{
    if(msg.orientation_covariance[0] == -1)
        return;
    tf::Quaternion quaternion(
        msg.orientation.x,
        msg.orientation.y,
        msg.orientation.z,
        msg.orientation.w
    );
    double roll, pitch, yaw;
    tf::Matrix3x3(quaternion).getRPY(roll, pitch, yaw);
    roll = roll * 180 / M_PI;
    pitch = pitch * 180 / M_PI;
    yaw = yaw * 180 / M_PI;
    ROS_INFO("roll:%f, pitch:%f, yaw:%f", roll, pitch, yaw);

    double target_yao = 90;
    double diff = target_yao - yaw;
    geometry_msgs::Twist vel_cmd;
    vel_cmd.angular.z = diff *0.01;
    vel_pub.publish(vel_cmd);   
    
}
int main(int argc, char  *argv[])
{
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "imu_node");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/imu/data", 1000, imu_callback);
    vel_pub = nh.advertise<geometry_msgs::Twist>("/cmd_vel", 1000);
    ros::spin();
    return 0;
}

File: src/imu_pkg/scripts/imu_node.py
#!/usr/bin/env python3
import rospy
from sensor_msgs.msg import Imu
from tf.transformations import euler_from_quaternion
import math
from geometry_msgs.msg import Twist 


def callback(msg):
    if msg.orientation_covariance[0] <0:
        return
    quatertion = [msg.orientation.x, msg.orientation.y, msg.orientation.z, msg.orientation.w]
    roll, pitch, yaw = euler_from_quaternion(quatertion)
    roll = math.degrees(roll)
    pitch = math.degrees(pitch)
    yaw = math.degrees(yaw)
    if yaw<0:
        yaw = yaw + 360
    print("roll: %f, pitch: %f, yaw: %f" % (roll, pitch, yaw))

    global vel_pub
    target_yaw = 180
    vel_cmd = Twist() 
    vel_cmd.angular.z = (target_yaw - yaw)/10 
    vel_pub.publish(vel_cmd)

    

if __name__ =='__main__':
    rospy.init_node('imu_node')
    rospy.Subscriber('imu/data', Imu, callback,queue_size=10)
    vel_pub =rospy.Publisher('cmd_vel', Twist, queue_size=10)
    rospy.spin()